src/networking.c

```diff
@@ -2762,7 +2769,7 @@ sds catClientInfoString(sds s, client *client) {
         getClientPeerId(client),
         getClientSockname(client),
         connGetInfo(client->conn, conninfo, sizeof(conninfo)),
-        client->name ? (char*)client->name->ptr : "",
+        (client->name && !shouldHideClientInfo()) ? (char*)client->name->ptr : "",
```

Comment:
Why should we hide the client name?

Reply:
client name can hold sensitive information about the client purpose/ what the client does

Reply:
> regarding client-list, I hide clients names because it is printed inside crash reports.
The client names will not be hidden when calling client-list directly.

ok, now i understand that change better (had difficulties wrapping my head around the code).

i think the solution of using `isBugReportStart` and `shouldHideClientInfo` is ugly, and instead, it would be much nicer to just add a `hide_user_info` argument to `catClientInfoString` and `getAllClientsInfoString`.
the problem with that is that `catClientInfoString` is widely used, and we don't wanna modify all the calls.
the solution to that is to add an alias function, i.e. `catClientInfoStringGeneric` will take the argument and the existing `catClientInfoString` will just be a wrapper.